src/test/java/com/aws/greengrass/tes/HttpServerImplTest.java

```diff
@@ -6,90 +6,340 @@
 package com.aws.greengrass.tes;
 
 import com.aws.greengrass.testcommons.testutilities.GGExtension;
-import com.sun.net.httpserver.HttpExchange;
+import com.aws.greengrass.testcommons.testutilities.TestUtils;
+import com.sun.net.httpserver.HttpContext;
 import com.sun.net.httpserver.HttpHandler;
+import com.sun.net.httpserver.HttpServer;
+import com.sun.net.httpserver.HttpsServer;
+import com.sun.net.httpserver.spi.HttpServerProvider;
+import lombok.RequiredArgsConstructor;
+import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.ExtensionContext;
 import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
 import org.junit.jupiter.params.provider.ValueSource;
-import org.mockito.Mock;
 import org.mockito.junit.jupiter.MockitoExtension;
 
+import javax.net.SocketFactory;
 import java.io.BufferedReader;
 import java.io.IOException;
-import java.io.InputStream;
 import java.io.InputStreamReader;
 import java.io.OutputStream;
 import java.net.HttpURLConnection;
+import java.net.Inet4Address;
+import java.net.Inet6Address;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.net.SocketAddress;
+import java.net.SocketException;
 import java.net.URL;
+import java.util.List;
+import java.util.concurrent.Executor;
 import java.util.concurrent.ExecutorService;
-import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
+import static com.aws.greengrass.testcommons.testutilities.ExceptionLogProtector.ignoreExceptionOfType;
 import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.junit.jupiter.api.Assertions.fail;
-import static org.mockito.Mockito.any;
-import static org.mockito.Mockito.doAnswer;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 
 @ExtendWith({MockitoExtension.class, GGExtension.class})
+@SuppressWarnings("PMD.AvoidUsingHardCodedIP")
 class HttpServerImplTest {
-    private static final String mockResponse = "Hello World";
 
-    private final ExecutorService executorService = Executors.newFixedThreadPool(1);
+    private static final String MOCK_CREDENTIAL_RESPONSE = "Hello World";
+    private static final String[] hosts = {"localhost", "127.0.0.1", "::1"};
 
-    @Mock
-    private HttpHandler mockHttpHandler;
+    ExecutorService executorService = TestUtils.synchronousExecutorService();
```

Comment:
shutdown this executor

Reply:
it's a no-op though? will add for completeness though